### DISCUSSION TITLE
Can we combine multiple GP's into a single additive model?

### DISCUSSION THREAD

**Discussion by kaijennissen** [GPflow/GPflow#1967] [Q&A] [answered]
I'd like to reproduce the [birthdays case study](https://avehtari.github.io/casestudies/Birthdays/birthdays.html).
The key idea is to model the relative number of births given per day $(y)$ using, amongst others, the following model:

$log(y)=intercept + f_{1}(x)+f_{2}(x)+exp(f_{3})\beta_{Day Of Week}+\beta_{Day Of Year}+\varepsilon$
$\varepsilon\sim Normal(0,\sigma)$
$intercept \sim Normal(0,1)$
$f_{1}\~\mathcal{GP}(0,K_{1})$
$f_{2}\~\mathcal{GP}(0,K_{2})$
$f_{3}\~\mathcal{GP}(0,K_{3})$
$\beta_{Day Of Week} = 0\, if\, Monday$
$\beta_{Day Of Week} \sim Normal(0,1)$
$\beta_{Day Of Year} \sim Normal(0,1)$

Is it possible to build such a model with `GPFlow`?

**Comment by st--** [comment, marked as answer]
In principle, yes; in practice, you might not want to:) You can build "simple" additive models where each component is a GP (including "parametric GPs" such as the intercept, which you can model with a Constant() kernel, or day-of-year and other special day effects for which you can build custom kernels that reflect the basis functions) by constructing a sum of kernels for each term in the sum. You can still build models that include non-linear effects such as the $\exp(f_3)$ but that becomes more involved (you have to construct an appropriate likelihood that combines all the GPs in whichever nonlinear way you want). You can do MCMC using the `GPMC` and `SGPMC` model classes together with tensorflow_probability (see [notebook](https://gpflow.github.io/GPflow/2.6.0/notebooks/advanced/mcmc.html)). That said, if you want to quickly iterate with different complex hierarchical models, Stan is probably more suitable!
